actors.md

```diff
@@ -326,11 +326,32 @@ func SlashConsensusFault(block1, block2 BlockHeader) {
 
 	miner := AuthorOf(block1)
 
-	// TODO: Some of the slashed collateral should be paid to the slasher
-
-	// Burn all of the miners collateral
-	miner.BurnCollateral()
-
+	// Check if miner has been slashed
+	// miner should be removed upon consensus fault slashing
+	if !self.Miners.Has(miner) {
+		Fatal("miner has been slashed")
+	}
+
+	// Burn miner pledge collateral
+	// TODO determine amount to slash if not the entire balance
+	// Tentatively, consensus fault should slash all pledge collateral
+	// only the balance is slashed if balance is less than collateral
+	// consensusFaultSlashAmount can return the Min for now
+	const requiredPledgeCollateral = PledgeCollateralForPower(miner.Power)
+	const slashedCollateral = consensusFaultSlashAmount(requiredPledgeCollateral, miner.Balance)
+
+	// Some of the slashed collateral should be paid to the slasher
+	// GROWTH_RATE determines how fast the slasher share of slashed collateral will increase as block elapses
+	// current GROWTH_RATE results in SLASHER_SHARE reaches 1 after 30 blocks
+	// TODO: parameter selection on GROWTH_RATE and INITIAL_SHARE
+	// TODO: define arithmetic precision and rounding for this operation
+	const blockElapsed = VM.CurrentBlockHeight() - block1.Height
+	const GROWTH_RATE = 1.26
+	const INITIAL_SHARE = 0.001
+	const SLASHER_SHARE = Min(Pow(INITIAL_SHARE, GROWTH_RATE), 1.0)
```

Comment:
I think this should be:
```
slasherSlash = min( INITIAL_SHARE * Pow(growthRate, blockElapsed), 1.0)
```
right?

Reply:
(blockElapsed should definitely be involved somehow)

Reply:
We're also going to have to be careful here, floating point precision and rounding needs to be very explicitly defined.

Reply:
Not exactly sure on the right floating point precision and rounding here though